uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -122,20 +140,79 @@ define([
         ]);
       })
       .then(initDates)
+      .then(initTimes)
       .then(initOriginalOpeningBalance)
       .then(function () {
         return $q.all([
           setDaySelectionMode(),
           initBalanceChange()
         ]);
       })
+      .then(initTimeAndDateInputsWatchers)
       .catch(handleError)
       .finally(function () {
         vm.loading.tab = false;
       });
     }());
 
     /**
+     * Amends balance change breakdown if calculation unit absence type is "hours".
+     * It changes the balance first and last days (only first day in case of single day request)
+     *   by setting the selected deductions.
+     * @NOTE this function mutates the "balanceChange" object
+     *
+     * @param  {Object} balanceChange
+     * @return {Object} amended balance change in case of "hours" absence type
+     */
+    function amendHourlyBalanceChangeBreakdown (balanceChange) {
+      var breakdown = balanceChange.breakdown;
+
+      if (!isCalculationUnit('hours')) {
+        return balanceChange;
+      }
+
+      if (balanceChange) {
+        amendHourlyBalanceChangeForDay(_.values(breakdown)[0], 'from');
```

Comment:
use `_.first(_.values(breakdown))` just for consistency with the statement below. It seems appropriate for this case.